File: ts/768/747.ts
function dominantIndex(nums: number[]): number {
  const len: number = nums.length
  if (len === 1) return 0
  let max: number = nums[0], maxIdx: number = 0, second: number = 0
  for (let i = 1; i < len; i++) {
    if (nums[i] > max) {
      second = max
      max = nums[i]
      maxIdx = i
    } else if (nums[i] > second) {
      second = nums[i]
    }
  }

  return max >= (second * 2) ? maxIdx : -1
}

File: c/768/747.c
int dominantIndex(int* nums, int numsSize){
  if (numsSize == 1) return 0;
  int maxIdx = 0, max = nums[0], second = 0;
  for (int i = 1; i < numsSize; i++) {
    if (nums[i] > max) {
      second = max;
      max = nums[i];
      maxIdx = i;
    } else if (nums[i] > second) {
      second = nums[i];
    }
  }
  return max >= (second * 2) ? maxIdx : -1;
}